Develop team pick pack ship entities

ian-a-hickey opened the pull request:
Opening this running pull request for daily review and touchbase.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ten24/slatwall/5312)
<!-- Reviewable:end -->

rmarchandTen24 commented:
need to address many-to-many linktable being over 30 characters. need to update tests to give exact feedback on that

rmarchandTen24 commented on model/entity/FulfillmentBatchItem.cfc:
```diff
@@ -0,0 +1,94 @@
+	// ============  END:  Non-Persistent Property Methods =================
+
+	// ============= START: Bidirectional Helper Methods ===================
+	
```
Need bidirectionals

rmarchandTen24 commented on model/entity/PickWave.cfc:
```diff
@@ -0,0 +1,104 @@
+	
+	// Fulfillment Batch Items (one-to-many)
+	public void function addFulfillmentBatchItem(required any fulfillmentBatchItem) {
+		arguments.fulfillmentBatchItem.setPickWave( this );
```
Can you create the bi-directionals on the other side

rmarchandTen24 commented on model/process/FulfillmentBatch_Create.cfc:
```diff
@@ -0,0 +1,61 @@
+component output="false" accessors="true" extends="HibachiProcess" {
+
+	// Injected Entities
+	property name="fulfillmentBatch";
```
Can you add the meta data for this as an object. It will help with the angular side

ian-a-hickey commented:
I shortened the link table name that was an issue. 
I updated all the bidirectionals to have both sides of the relationship.
I added some helper methods to fulfillmentBatch to make it easy to get some status data.

rmarchandTen24 commented on model/entity/Account.cfc:
```diff
@@ -96,7 +96,9 @@ component displayname="Account" entityname="SlatwallAccount" table="SwAccount" p
 	property name="termAccountOrderPayments" singularname="termAccountOrderPayment" cfc="OrderPayment" type="array" fieldtype="one-to-many" fkcolumn="termPaymentAccountID" cascade="all" inverse="true";
 	property name="giftCards" singularname="giftCard" cfc="GiftCard" type="array" fieldtype="one-to-many" fkcolumn="ownerAccountID" cascade="all" inverse="true";
-
+	property name="fulfillmentBatches" singularname="fulfillmentBatch" fieldType="one-to-many" type="array" fkColumn="accountID" cfc="fulfillmentBatch" inverse="true" orderby="createdDateTime asc";
```
CFC needs to be capitalized

rmarchandTen24 replied:
Please remove the orderby. We enforce that via smart list.

ian-a-hickey replied:
removed the orderBy and capitalized

rmarchandTen24 commented on model/entity/OrderDeliveryItem.cfc:
```diff
@@ -59,7 +59,8 @@ component displayname="Order Delivery Item" entityname="SlatwallOrderDeliveryIte
 	// Related Object Properties (one-to-many)
 	property name="referencingOrderItems" singularname="referencingOrderItem" cfc="OrderItem" fieldtype="one-to-many" fkcolumn="referencedOrderDeliveryItemID" inverse="true" cascade="all"; // Used For Returns
-
+	property name="fulfillmentBatchItems" singularname="fulfillmentBatchItem" fieldType="one-to-many" type="array" fkColumn="orderDeliveryItemID" cfc="fulfillmentBatchItem" inverse="true" orderby="createdDateTime asc";
```
Capitalize cfc

rmarchandTen24 replied:
Remove order y

rmarchandTen24 commented on model/entity/OrderItem.cfc:
```diff
@@ -78,7 +78,8 @@ component entityname="SlatwallOrderItem" table="SwOrderItem" persistent="true" a
 	property name="giftCards" singularname="giftCard" cfc="GiftCard" type="array" fieldtype="one-to-many" fkcolumn="originalOrderItemID" cascade="all" inverse="true";
 	property name="orderItemGiftRecipients" singularname="orderItemGiftRecipient" cfc="OrderItemGiftRecipient" type="array" fieldtype="one-to-many" fkcolumn="orderItemID" cascade="all" inverse="true";
-
+	property name="fulfillmentBatchItems" singularname="fulfillmentBatchItem" fieldType="one-to-many" type="array" fkColumn="orderItemID" cfc="fulfillmentBatchItem" inverse="true" orderby="createdDateTime asc";
```
CFC capitalize. Remove orderby

rmarchandTen24 commented on model/entity/Stock.cfc:
```diff
@@ -58,7 +58,8 @@ component displayname="Stock" entityname="SlatwallStock" table="SwStock" persist
 	property name="vendorOrderItems" singularname="vendorOrderItem" cfc="VendorOrderItem" fieldtype="one-to-many" fkcolumn="stockID" inverse="true";
 	property name="inventory" singularname="inventory" cfc="Inventory" fieldtype="one-to-many" fkcolumn="stockID" inverse="true" lazy="extra";
-
+	property name="fulfillmentBatchItems" singularname="fulfillmentBatchItem" fieldType="one-to-many" type="array" fkColumn="stockID" cfc="fulfillmentBatchItem" inverse="true" orderby="createdDateTime asc";
```
CFC capitalize. Remove orderby

rmarchandTen24 commented on model/entity/Account.cfc:
```diff
@@ -96,7 +96,9 @@ component displayname="Account" entityname="SlatwallAccount" table="SwAccount" p
 	property name="giftCards" singularname="giftCard" cfc="GiftCard" type="array" fieldtype="one-to-many" fkcolumn="ownerAccountID" cascade="all" inverse="true";
-
+	property name="fulfillmentBatches" singularname="fulfillmentBatch" fieldType="one-to-many" type="array" fkColumn="accountID" cfc="fulfillmentBatch" inverse="true" orderby="createdDateTime asc";
+	property name="pickWaves" singularname="pickWave" fieldType="one-to-many" type="array" fkColumn="accountID" cfc="PickWave" inverse="true" orderby="createdDateTime asc";
```
Remove orderby

rmarchandTen24 commented on model/entity/OrderFulfillment.cfc:
```diff
@@ -73,7 +73,8 @@ component displayname="Order Fulfillment" entityname="SlatwallOrderFulfillment"
 	property name="accountLoyaltyTransactions" singularname="accountLoyaltyTransaction" cfc="AccountLoyaltyTransaction" type="array" fieldtype="one-to-many" fkcolumn="orderFulfillmentID" cascade="all" inverse="true";
 	property name="attributeValues" singularname="attributeValue" cfc="AttributeValue" type="array" fieldtype="one-to-many" fkcolumn="orderFulfillmentID" cascade="all-delete-orphan" inverse="true";
-
+	property name="fulfillmentBatchItems" singularname="fulfillmentBatchItem" fieldType="one-to-many" type="array" fkColumn="orderFulfillmentID" cfc="fulfillmentBatchItem" inverse="true" orderby="createdDateTime asc";
```
Capitalize cfc. Remove orderby
